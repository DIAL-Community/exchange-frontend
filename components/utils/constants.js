export const DisplayType = {
  LARGE_CARD: 'large-card',
  SMALL_CARD: 'small-card',
  PINNED_CARD: 'pinned-card',
  FEATURED_CARD: 'featured-card',
  HUB_CARD: 'hub-card',
  GRID_CARD: 'grid-card'
}

export const MainDisplayType = {
  LIST: 'list',
  GRID: 'grid'
}

export const ORIGIN_SLUG_EXPANSIONS = {
  'dial': 'Digital Impact Alliance',
  'digital-government-platform-tracker': 'Digital Government Platform Tracker',
  'digital-health-atlas': 'Digital Health Atlas',
  'digital-square': 'Digital Square',
  'dpga': 'Digital Public Goods Alliance',
  'giz': 'GIZ',
  'indiastack': 'India Stack',
  'manually-entered': 'Manually Entered',
  'unicef-covid': 'UNICEF'
}

export const OpportunityStatus = {
  CLOSED: 'CLOSED',
  OPEN: 'OPEN',
  UPCOMING: 'UPCOMING'
}

export const OpportunityType = {
  BID: 'BID',
  BUILDING_BLOCK: 'BUILDING BLOCK',
  INNOVATION: 'INNOVATION',
  TENDER: 'TENDER',
  OTHER: 'OTHER'
}

export const MaturityStatus = {
  DRAFT: 'DRAFT',
  PUBLISHED: 'PUBLISHED'
}

export const CategoryType = {
  DPI: 'DPI',
  FUNCTIONAL: 'FUNCTIONAL'
}

export const MappingStatus = {
  BETA: 'BETA',
  MATURE: 'MATURE',
  SELF_REPORTED: 'SELF-REPORTED',
  VALIDATED: 'VALIDATED'
}

export const LicenseTypeFilter = {
  ALL: 'all_license',
  COMMERCIAL: 'commercial_only',
  OPEN_SOURCE:'oss_only'
}

export const ObjectType = {
  BUILDING_BLOCK: 'BUILDING_BLOCK',
  DATASET: 'OPEN_DATA',
  PRODUCT: 'PRODUCT',
  PROJECT: 'PROJECT',
  OPPORTUNITY: 'OPPORTUNITY',
  ORGANIZATION: 'ORGANIZATION',
  USE_CASE: 'USE_CASE',
  WORKFLOW: 'WORKFLOW',
  MOVE: 'MOVE',
  PLAY: 'PLAY',
  PLAYBOOK: 'PLAYBOOK',
  RUBRIC_CATEGORY: 'RUBRIC_CATEGORY',
  CATEGORY_INDICATOR: 'CATEGORY_INDICATOR',
  CANDIDATE_DATASET: 'CANDIDATE_OPEN_DATA',
  CANDIDATE_ORGANIZATION: 'CANDIDATE_ORGANIZATION',
  CANDIDATE_PRODUCT: 'CANDIDATE_PRODUCT',
  CANDIDATE_RESOURCE: 'CANDIDATE_RESOURCE',
  CANDIDATE_ROLE: 'CANDIDATE_ROLE',
  TAG: 'TAG',
  SECTOR: 'SECTOR',
  URL: 'URL',
  CITY: 'CITY',
  COUNTRY: 'COUNTRY',
  REGION: 'REGION',
  TASK: 'TASK',
  USER: 'USER',
  CONTACT: 'CONTACT',
  RESOURCE: 'RESOURCE',
  RESOURCE_TOPIC: 'RESOURCE_TOPIC',
  SITE_SETTING: 'SITE_SETTING',
  TENANT_SETTING: 'TENANT_SETTING'

}

export const CategoryIndicatorType = {
  NUMERIC: 'numeric',
  SCALE: 'scale',
  BOOLEAN: 'boolean'
}

export const CandidateActionType = {
  APPROVE: 'APPROVE',
  REJECT: 'REJECT'
}

export const DEFAULT_PAGE_SIZE = 8

export const ProductStageType = {
  PILOT: 'pilot',
  SCALING: 'scaling',
  MATURE: 'mature'
}

export const ProductExtraAttributeNames = [
  'Relevance',
  'Impact',
  'Local Ownership',
  'Funders'
]

export const ProductMaturityAttributeNames = [
  'Deployments',
  'Deployed Countries',
  'Active Users',
  'Transactions per month',
  'Annual Revenue',
  'Funding Raised'
]
