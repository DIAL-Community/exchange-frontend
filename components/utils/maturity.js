export const getCategoryIndicatorTypes = (format) => [
  { label: format('shared.categoryIndicator.type.numeric'), value: 'numeric' },
  { label: format('shared.categoryIndicator.type.scale'), value: 'scale' },
  { label: format('shared.categoryIndicator.type.boolean'), value: 'boolean' }
]

export const getCategoryIndicatorScaleOptions = (format) => [
  { label: format('shared.categoryIndicator.type.scale.low'), value: 'low' },
  { label: format('shared.categoryIndicator.type.scale.medium'), value: 'medium' },
  { label: format('shared.categoryIndicator.type.scale.high'), value: 'high' }
]

export const getCategoryIndicatorBooleanOptions = (format) => [
  { label: format('shared.categoryIndicator.type.boolean.true'), value: 't' },
  { label: format('shared.categoryIndicator.type.boolean.false'), value: 'f' }
]

export const getCategoryIndicatorNumericOptions = () =>
  ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10'].map(value => ({ label: value, value }))
