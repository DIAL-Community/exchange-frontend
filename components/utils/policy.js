import { GRAPH_QUERY_CONTEXT_KEY, GRAPH_QUERY_POLICY_SLUG } from '../../lib/apolloClient'

const operationAllowed = (error, operationName) => {
  if (error?.graphQLErrors) {
    const { graphQLErrors } = error
    if (graphQLErrors.length > 0) {
      graphQLErrors.forEach(e => {
        if (e.extensions.operation === operationName) {
          return true
        }
      })
    }
  }

  return false
}

// Return map of operation and boolean value for each operation
// e.g. { editing: true, deleting: true }
export const fetchOperationPolicies = async (client, query, operations, variables = { slug: GRAPH_QUERY_POLICY_SLUG }) => {
  const response = await client.query({
    query,
    variables,
    errorPolicy: 'all',
    context: {
      headers: {
        [GRAPH_QUERY_CONTEXT_KEY]: operations.join(', ')
      }
    }
  })

  const policies = {}
  operations.forEach(operation => {
    policies[operation] = operationAllowed(response.error, operation)
  })

  return policies
}
