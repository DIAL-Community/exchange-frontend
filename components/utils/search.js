import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'

const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({
    query,
    variables,
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMaturity = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    mature: true,
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({
    query,
    variables,
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  return response.data ? fetchedDataCallback(response.data) : []
}
