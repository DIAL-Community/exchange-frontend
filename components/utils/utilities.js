export const prependUrlWithProtocol = (url) => '//' + url

export const isValidFn = (maybeFunction) =>
  maybeFunction && {}.toString.call(maybeFunction) === '[object Function]'

export const isDebugLoggingEnabled = () => String(process.env.NEXT_PUBLIC_ENABLE_DEBUG).toLowerCase() === 'true'
