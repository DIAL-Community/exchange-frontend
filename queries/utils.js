const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
