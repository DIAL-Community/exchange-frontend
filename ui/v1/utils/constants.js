export const DisplayType = {
  LARGE_CARD: 'large-card',
  SMALL_CARD: 'small-card',
  PINNED_CARD: 'pinned-card'
}

export const OpportunityStatus = {
  CLOSED: 'CLOSED',
  OPEN: 'OPEN',
  UPCOMING: 'UPCOMING'
}

export const OpportunityType = {
  BID: 'BID',
  BUILDING_BLOCK: 'BUILDING BLOCK',
  INNOVATION: 'INNOVATION',
  TENDER: 'TENDER',
  OTHER: 'OTHER'
}

export const MaturityStatus = {
  DRAFT: 'DRAFT',
  PUBLISHED: 'PUBLISHED'
}

export const CategoryType = {
  DPI: 'DPI',
  FUNCTIONAL: 'FUNCTIONAL'
}

export const MappingStatus = {
  BETA: 'BETA',
  MATURE: 'MATURE',
  SELF_REPORTED: 'SELF-REPORTED',
  VALIDATED: 'VALIDATED'
}

export const LicenseTypeFilter = {
  ALL: 'all_license',
  COMMERCIAL: 'commercial_only',
  OPEN_SOURCE:'oss_only'
}

export const ObjectType = {
  BUILDING_BLOCK: 'BUILDING_BLOCK',
  DATASET: 'OPEN_DATA',
  PRODUCT: 'PRODUCT',
  PROJECT: 'PROJECT',
  OPPORTUNITY: 'OPPORTUNITY',
  ORGANIZATION: 'ORGANIZATION',
  PLAYBOOK: 'PLAYBOOK',
  USE_CASE: 'USE_CASE',
  CANDIDATE_DATASET: 'CANDIDATE_OPEN_DATA',
  CANDIDATE_ORGANIZATION: 'CANDIDATE_ORGANIZATION',
  CANDIDATE_PRODUCT: 'CANDIDATE_PRODUCT',
  CANDIDATE_ROLE: 'CANDIDATE_ROLE',
  TAG: 'TAG',
  SECTOR: 'SECTOR',
  COUNTRY: 'COUNTRY',
  CITY: 'CITY',
  CONTACT: 'CONTACT',
  URL: 'URL',
  PLAY: 'PLAY',
  MOVE: 'MOVE'
}

export const CategoryIndicatorType = {
  NUMERIC: 'numeric',
  SCALE: 'scale',
  BOOLEAN: 'boolean'
}

export const CandidateActionType = {
  APPROVE: 'APPROVE',
  REJECT: 'REJECT'
}

export const DEFAULT_PAGE_SIZE = 8
