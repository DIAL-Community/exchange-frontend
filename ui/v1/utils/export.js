import { saveAs } from 'file-saver'

export const ExportType = {
  EXPORT_AS_CSV: 'csv',
  EXPORT_AS_JSON: 'json'
}

/* Convert the object keys to snake case key because rails is using snake case. */
export const convertKeys = (object) => {
  Object.keys(object).forEach(key => {
    // Flatten the filter selection if it's Array.
    // Convert it to array of slug only.
    if (Array.isArray(object[key])) {
      object[key] = object[key].map(value => value.slug)
    }

    // Convert the key to snake case.
    const snakeCaseKey = key.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`)
    if (key !== snakeCaseKey) {
      Object.defineProperty(object, snakeCaseKey,
        Object.getOwnPropertyDescriptor(object, key))
      delete object[key]
    }
  })

  return object
}

export const asyncExport = (exportType, backendType, exportParameters, userEmail) => {
  const exportPath = process.env.NEXT_PUBLIC_AUTH_SERVER + `/api/v1/${backendType}.${exportType}`
  fetch(
    exportPath,
    {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
        'X-Requested-With': 'XMLHttpRequest',
        'Access-Control-Allow-Origin': process.env.NEXT_PUBLIC_AUTH_SERVER,
        'Access-Control-Allow-Credentials': true,
        'Access-Control-Allow-Headers': 'Set-Cookie',
        'X-User-Email': userEmail
      },
      body: JSON.stringify(exportParameters)
    }
  )
    .then(response => response.body)
    .then(body => {
      const reader = body.getReader()

      return new ReadableStream({
        start (controller) {
          return pump()
          async function pump () {
            const { done, value } = await reader.read()
            // When no more data needs to be consumed, close the stream
            if (done) {
              controller.close()

              return
            }

            // Enqueue the next data chunk into our target stream
            controller.enqueue(value)

            return pump()
          }
        }
      })
    })
    .then(stream => new Response(stream))
    .then(response => response.blob())
    .then(blob => {
      saveAs(blob, `${backendType}-data.${exportType}`)
    })
}
