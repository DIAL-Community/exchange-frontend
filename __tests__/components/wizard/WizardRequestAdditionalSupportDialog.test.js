import { fireEvent, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import AdditionalSupportDialog from '../../../components/wizard/AdditionalSupportDialog'
import { mockObserverImplementation, render, waitForAllEffects } from '../../test-utils'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import CustomMockedProvider from '../../utils/CustomMockedProvider'

mockNextUseRouter()

describe('Unit test for the AdditionalSupportDialog component.', () => {
  const SUBMIT_BUTTON_TEST_ID = 'submit-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const NAME_TEST_ID = 'name'
  const EMAIL_ADDRESS_TEST_ID = 'email-address'
  const MESSAGE_TEST_ID = 'message'
  const mockIsDialogOpen = true
  const mockSetIsDialogOpen = jest.fn()
  const DIALOG_TEST_ID = 'dialog'
  const REQUIRED_FIELD_MESSAGE = 'This field is required'
  const INVALID_EMAIL_ADDRESS_MESSAGE = 'Please enter a valid email address'
  const MESSAGE_TOO_SHORT_MESSAGE = 'Message must be at least 20 characters'

  beforeAll(() => {
    window.ResizeObserver = mockObserverImplementation()
    window.IntersectionObserver = mockObserverImplementation()
  })

  test('Should match snapshot.', () => {
    const { getByTestId } = render(
      <CustomMockedProvider>
        <AdditionalSupportDialog
          isOpen={mockIsDialogOpen}
          onClose={mockSetIsDialogOpen}
        />
      </CustomMockedProvider>
    )
    expect(getByTestId(DIALOG_TEST_ID)).toMatchSnapshot()
  })

  test('Should call the onClose function after clicking the "Cancel" button.', () => {
    const { getByTestId } = render(
      <CustomMockedProvider>
        <AdditionalSupportDialog
          isOpen={mockIsDialogOpen}
          onClose={mockSetIsDialogOpen}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
    expect(mockSetIsDialogOpen).toHaveBeenCalled()
  })

  test('Should display validation errors after clicking the "Confirm" button.', async () => {
    const user = userEvent.setup()
    const { getByTestId } = render(
      <CustomMockedProvider>
        <AdditionalSupportDialog
          isOpen={mockIsDialogOpen}
          onClose={mockSetIsDialogOpen}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(SUBMIT_BUTTON_TEST_ID))
    await waitForAllEffects(1000)
    expect(getByTestId(NAME_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(EMAIL_ADDRESS_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(MESSAGE_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(DIALOG_TEST_ID)).toMatchSnapshot()

    await user.type(screen.getByLabelText(/Name/), 'test name')
    expect(getByTestId(NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await user.type(screen.getByLabelText(/Email Address/), 'test email address')
    expect(getByTestId(EMAIL_ADDRESS_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(EMAIL_ADDRESS_TEST_ID)).toHaveTextContent(INVALID_EMAIL_ADDRESS_MESSAGE)

    await user.clear(screen.getByLabelText(/Email Address/))
    await user.type(screen.getByLabelText(/Email Address/), 'test@email.address')
    expect(getByTestId(EMAIL_ADDRESS_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(EMAIL_ADDRESS_TEST_ID)).not.toHaveTextContent(INVALID_EMAIL_ADDRESS_MESSAGE)

    await user.type(screen.getByLabelText(/Message/), 'test message')
    expect(getByTestId(MESSAGE_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(MESSAGE_TEST_ID)).toHaveTextContent(MESSAGE_TOO_SHORT_MESSAGE)

    await user.type(screen.getByLabelText(/Message/), 'test message')
    expect(getByTestId(MESSAGE_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(MESSAGE_TEST_ID)).not.toHaveTextContent(MESSAGE_TOO_SHORT_MESSAGE)
  })
})
