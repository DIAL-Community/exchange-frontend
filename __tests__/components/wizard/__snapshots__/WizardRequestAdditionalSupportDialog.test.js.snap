// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the AdditionalSupportDialog component. Should display validation errors after clicking the "Confirm" button. 1`] = `
<div
  aria-describedby="headlessui-description-:r8:"
  aria-modal="true"
  class="fixed z-100"
  data-headlessui-state="open"
  data-testid="dialog"
  id="headlessui-dialog-:r6:"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center"
  >
    <div
      class="w-4/5 sm:w-4/5 md:w-3/5 lg:w-2/5 xl:w-2/6 p-2 sm:p-4 transform bg-white shadow-xl rounded-2xl flex flex-col opacity-100 scale-100"
      data-headlessui-state="open"
      id="headlessui-dialog-panel-:r7:"
    >
      <div
        class="bg-white overflow-auto py-4 px-8 flex flex-col gap-3"
        data-headlessui-state="open"
        id="headlessui-description-:r8:"
      >
        <div
          class="flex justify-start text-xl"
          data-testid="dialog-body"
        >
          <div
            class="w-full"
          >
            <form
              id="wizard-request-additional-support-form"
            >
              <div
                class="pb-12 mb-4 flex flex-col gap-3"
              >
                <div
                  class="text-2xl font-semibold text-dial-sapphire pb-4"
                >
                  Request additional support
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="name"
                >
                  <label
                    class="form-field-label required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="validation-error w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                  <p
                    class="text-sm text-validation-error"
                  >
                    This field is required
                  </p>
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="email-address"
                >
                  <label
                    class="form-field-label required-field"
                    for="email-address"
                  >
                    Email Address
                  </label>
                  <input
                    class="validation-error w-full"
                    id="email-address"
                    name="emailAddress"
                    placeholder="Email Address"
                    type="text"
                    value=""
                  />
                  <p
                    class="text-sm text-validation-error"
                  >
                    This field is required
                  </p>
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="message"
                >
                  <label
                    class="form-field-label required-field"
                    for="message"
                  >
                    Message
                  </label>
                  <textarea
                    class="validation-error textarea w-full"
                    id="message"
                    name="message"
                    placeholder="Message"
                    rows="6"
                  />
                  <p
                    class="text-sm text-validation-error"
                  >
                    This field is required
                  </p>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="flex justify-start gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            form="wizard-request-additional-support-form"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the AdditionalSupportDialog component. Should match snapshot. 1`] = `
<div
  aria-describedby="headlessui-description-:r2:"
  aria-modal="true"
  class="fixed z-100"
  data-headlessui-state="open"
  data-testid="dialog"
  id="headlessui-dialog-:r0:"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center"
  >
    <div
      class="w-4/5 sm:w-4/5 md:w-3/5 lg:w-2/5 xl:w-2/6 p-2 sm:p-4 transform bg-white shadow-xl rounded-2xl flex flex-col"
      data-headlessui-state="open"
      id="headlessui-dialog-panel-:r1:"
    >
      <div
        class="bg-white overflow-auto py-4 px-8 flex flex-col gap-3"
        data-headlessui-state="open"
        id="headlessui-description-:r2:"
      >
        <div
          class="flex justify-start text-xl"
          data-testid="dialog-body"
        >
          <div
            class="w-full"
          >
            <form
              id="wizard-request-additional-support-form"
            >
              <div
                class="pb-12 mb-4 flex flex-col gap-3"
              >
                <div
                  class="text-2xl font-semibold text-dial-sapphire pb-4"
                >
                  Request additional support
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="name"
                >
                  <label
                    class="form-field-label required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="email-address"
                >
                  <label
                    class="form-field-label required-field"
                    for="email-address"
                  >
                    Email Address
                  </label>
                  <input
                    class="w-full"
                    id="email-address"
                    name="emailAddress"
                    placeholder="Email Address"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="message"
                >
                  <label
                    class="form-field-label required-field"
                    for="message"
                  >
                    Message
                  </label>
                  <textarea
                    class="textarea w-full"
                    id="message"
                    name="message"
                    placeholder="Message"
                    rows="6"
                  />
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="flex justify-start gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            form="wizard-request-additional-support-form"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
